go-backend/db/organization.go

```diff
@@ -0,0 +1,216 @@
+		updated_on FROM organizations WHERE id=$1`
+	
+	ListOrganizationsQuery =`SELECT id,
+		name,
+		contact_email,
+		domain_name,
+		subscription_status,
+		subscription_valid_upto,
+		hi5_limit,
+		hi5_quota_renewal_frequency,
+		timezone,
+		created_by,
+		created_on,
+		updated_by,
+		updated_on FROM organizations ORDER BY name ASC`
+	emailRegex = 	"^[a-zA-Z0-9.!#$%&'*+/=?^_`{|}~-]+@[a-zA-Z0-9](?:[a-zA-Z0-9-]{0,61}[a-zA-Z0-9])?(?:\\.[a-zA-Z0-9](?:[a-zA-Z0-9-]{0,61}[a-zA-Z0-9])?)*$"
+	domainRegex = `(?:[a-z0-9](?:[a-z0-9-]{0,61}[a-z0-9])?\.)+[a-z0-9][a-z0-9-]{0,61}[a-z0-9]`
+)
+
+type Organization struct {
+	ID                       int       `db:"id" json:"id" `
+	Name                     string    `db:"name" json:"name"`
+	ContactEmail             string    `db:"contact_email" json:"email"`
+	DomainName               string    `db:"domain_name" json:"domain_name"`
+	SubscriptionStatus       int       `db:"subscription_status" json:"subscription_status"`
+	SubscriptionValidUpto    time.Time `db:"subscription_valid_upto" json:"subscription_valid_upto"`
+	Hi5Limit                 int    `db:"hi5_limit" json:"hi5_limit"`
+	Hi5QuotaRenewalFrequency string    `db:"hi5_quota_renewal_frequency" json:"hi5_quota_renewal_frequency"`
+	Timezone                 string    `db:"timezone" json:"timezone"`
+	CreatedBy                int    `db:"created_by" json:"_"`
+	CreatedOn                time.Time `db:"created_on" json:"_"`
+	UpdatedBy                int    `db:"updated_by" json:"_"`
+	UpdatedOn                time.Time `db:"updated_on" json:"_"`
+}
+
+//TODO how to declare this as reusable
+type ErrorResponse struct {
+	Code 	string `json:"code"`
+	Message	string `json:"message"`
+	Fields map[string]string `json:"fields"`
+}
+
+func (org *Organization) ValidateOrganization() (errorResponse map[string]ErrorResponse, valid bool) {
+
+	valid = true
+	fieldErrors := make(map[string]string)
+	var validEmail, _ = regexp.Compile(emailRegex)
+	var validDomain, _ = regexp.Compile(domainRegex)
+	
+	if org.Name == "" {
+		valid = false
+		fieldErrors["name"] = "Can't be blank"
+	}
+
+	if !validEmail.MatchString(org.ContactEmail) {
+		valid = false
+		fieldErrors["email"] = "Please enter a valid email"
+	}
+
+	if !validDomain.MatchString(org.DomainName) {
+		fieldErrors["domain_name"] = "Please enter valid domain"
+		valid = false
+	}
+
+	errorResponse = map[string]ErrorResponse{"error": ErrorResponse{
+		Code: "invalid_data",
+		Message: "Please provide valid organization data",
+		Fields: fieldErrors,
+	},
+}
+
+	//TODO: Ask what other validations are expected
+	
+	return
+}
+
+func (s *pgStore) ListOrganizations(ctx context.Context) (organizations []Organization, err error) {
+
+	err = s.db.Select(&organizations, ListOrganizationsQuery)
+	if err != nil {
+		logger.WithField("err", err.Error()).Error("Error listing organizations")
+		return
+	}
+
+	return
+}
+
+func (s *pgStore) CreateOrganization(ctx context.Context, org Organization) (err error) {
```

Comment:
@mayuriardad @rohitjoshixyz 

we also need to write db integration test cases, which will actually make a DB call and test these db functions.
Remember the calls should go to a test database.